src/servers/http_server.cc

```diff
@@ -624,791 +458,140 @@ HTTPAPIServer::HandleStatus(evhtp_request_t* req, const std::string& status_uri)
-               ? EVHTP_RES_OK
-               : EVHTP_RES_BADREQ);
-
+  h2o_start_response(req, &generator);
+  h2o_send(req, &body, 1, H2O_SEND_STATE_FINAL);
   TRTSERVER_ErrorDelete(err);
-}
-
-void
-HTTPAPIServer::HandleModelControl(
-    evhtp_request_t* req, const std::string& modelcontrol_uri)
-{
-  if (req->method != htp_method_POST) {
-    evhtp_send_reply(req, EVHTP_RES_METHNALLOWED);
-    return;
-  }
 
-  std::string action_type_str, model_name;
-  if ((modelcontrol_uri.empty()) || (!RE2::FullMatch(
-                                        modelcontrol_uri, modelcontrol_regex_,
-                                        &action_type_str, &model_name))) {
-    evhtp_send_reply(req, EVHTP_RES_BADREQ);
-    return;
-  }
-
-  TRTSERVER_Error* err = nullptr;
-  if (action_type_str == "load") {
-    err = TRTSERVER_ServerLoadModel(server_.get(), model_name.c_str());
-  } else if (action_type_str == "unload") {
-    err = TRTSERVER_ServerUnloadModel(server_.get(), model_name.c_str());
-  } else {
-    err = TRTSERVER_ErrorNew(
-        TRTSERVER_ERROR_UNKNOWN,
-        std::string("unknown action type '" + action_type_str + "'").c_str());
-  }
-
-  RequestStatus request_status;
-  RequestStatusUtil::Create(
-      &request_status, err, RequestStatusUtil::NextUniqueRequestId(),
-      server_id_);
-
-  evhtp_headers_add_header(
-      req->headers_out,
-      evhtp_header_new(
-          kStatusHTTPHeader, request_status.ShortDebugString().c_str(), 1, 1));
-
-  evhtp_send_reply(
-      req, (request_status.code() == RequestStatusCode::SUCCESS)
-               ? EVHTP_RES_OK
-               : EVHTP_RES_BADREQ);
-
-  TRTSERVER_ErrorDelete(err);
+  return 0;
 }
 
-#ifdef TRTIS_ENABLE_GPU
 TRTSERVER_Error*
-HTTPAPIServer::EVBufferToCudaHandle(
-    evbuffer* handle_buffer, cudaIpcMemHandle_t** cuda_shm_handle)
+HTTPAPIServer::Start()
 {
-  // Extract serialzied cuda IPC handle from HTTP body and store in
-  // 'cuda_shm_handle'.
-  struct evbuffer_iovec* v = nullptr;
-  *cuda_shm_handle = nullptr;
-  size_t byte_size = sizeof(cudaIpcMemHandle_t);
-
-  int n = evbuffer_peek(handle_buffer, -1, NULL, NULL, 0);
-  if (n > 0) {
-    v = static_cast<struct evbuffer_iovec*>(
-        alloca(sizeof(struct evbuffer_iovec) * n));
-    if (evbuffer_peek(handle_buffer, -1, NULL, v, n) != n) {
-      return TRTSERVER_ErrorNew(
-          TRTSERVER_ERROR_INTERNAL, "unexpected error getting input buffers ");
-    }
-  }
+  h2o_hostconf_t* hostconf;
+  h2o_pathconf_t* pathconf;
+  h2o_access_log_filehandle_t* logfh = h2o_access_log_open_handle(
+      "/dev/stdout", NULL, H2O_LOGCONF_ESCAPE_APACHE);
 
-  if (byte_size != v[0].iov_len) {
-    return TRTSERVER_ErrorNew(
-        TRTSERVER_ERROR_INVALID_ARG,
-        std::string(
-            "unexpected size for CUDA shared-memory handle, expecting " +
-            std::to_string(byte_size) + " bytes")
-            .c_str());
-  }
+  signal(SIGPIPE, SIG_IGN);
```

Comment:
Why ignore sigpipe?

Reply:
"You generally want to ignore the SIGPIPE and handle the error directly in your code. This is because signal handlers in C have many restrictions on what they can do. The most portable way to do this is to set the SIGPIPE handler to SIG_IGN ."

This is the recommended way suggested by h2o. But I can read further and remove it if not needed.

Reply:
It is needed so h2o can capture and proceed to kill+cleanup the server accordingly.